quantum/process_keycode/process_autocorrect.c

```diff
@@ -19,6 +19,30 @@
 #    include "autocorrect_data_default.h"
 #endif
 
+#ifndef AUTOCORRECT_LINK_BYTE_COUNT
+#    define AUTOCORRECT_LINK_BYTE_COUNT 2
+#endif
+
+#if AUTOCORRECT_LINK_BYTE_COUNT < 2 || AUTOCORRECT_LINK_BYTE_COUNT > 3
+#    error "AUTOCORRECT_LINK_BYTE_COUNT must be either 2 or 3 bytes."
+#endif
+
+#if defined(__AVR__) && DICTIONARY_SIZE > UINT16_MAX
```

Comment:
We would not accept platform specific logic in core, when there is an abstraction in place that already exists.

`platforms/progmem.h` would need to be updated to add `pgm_read_byte_far` compatibility.

```suggestion
#if DICTIONARY_SIZE > UINT16_MAX
```